packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx

```diff
@@ -63,100 +64,259 @@ type ExpandableBlockProps = {
+      as='button'
+      type='button'
+      className={cx('iui-expandable-header', className)}
+      aria-expanded={isExpanded}
+      aria-disabled={disabled}
+      tabIndex={0}
+      onClick={mergeEventHandlers(onClickProp, () => {
+        if (disabled) {
+          return;
+        }
+        setExpanded(!isExpanded);
+        onToggle?.(!isExpanded);
+      })}
+      ref={forwardedRef}
+      {...rest}
+    >
+      {children ?? (
+        <>
+          {expandIcon && <ExpandableBlock.ExpandIcon />}
+          <ExpandableBlock.LabelArea>
+            <ExpandableBlock.Title>{label}</ExpandableBlock.Title>
+          </ExpandableBlock.LabelArea>
+        </>
       )}
-      style={style}
-      ref={ref}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'button', ExpandableBlockHeaderOwnProps>;
+ExpandableBlockHeader.displayName = 'ExpandableBlock.Header';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.ExpandIcon component
+
+const ExpandableBlockExpandIcon = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-expandable-block-icon', className)}
+      ref={forwardedRef}
       {...rest}
     >
+      {children ?? <SvgChevronRight aria-hidden />}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+ExpandableBlockExpandIcon.displayName = 'ExpandableBlock.ExpandIcon';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.LabelArea component
+
+const ExpandableBlockLabelArea = polymorphic.span('iui-expandable-block-label');
+ExpandableBlockLabelArea.displayName = 'ExpandableBlock.LabelArea';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Title component
+
+const ExpandableBlockTitle = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-expandable-block-title', className)}
+      ref={forwardedRef}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+ExpandableBlockTitle.displayName = 'ExpandableBlock.Title';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Caption component
+
+const ExpandableBlockCaption = polymorphic('iui-expandable-block-caption');
+ExpandableBlockCaption.displayName = 'ExpandableBlock.Caption';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.EndIcon component
+
+const ExpandableBlockEndIcon = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  const { status } = useSafeContext(ExpandableBlockContext);
+
+  const icon = children ?? (status && StatusIconMap[status]());
+  return (
+    <Icon className={className} fill={status} ref={forwardedRef} {...rest}>
```

Comment:
if there is no additional class here, then `className` doesn't need to be destructed, it will be used automatically through `...rest` props.

also, `iui-status-icon` can be removed from scss/html code

Reply:
thanks, updated and removed